hack/logextractor.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+#
+# This script can be used to extract the failure cluster logs tarball from the PVC created for an install pod.
+#
+# Usage: ./logextractor.sh [CLUSTER_DEPLOYMENT_NAME] [DEST_DIR]
+#
+
+set -e
+
+POD_NAME="hive-logextractor"
+PVC_NAME=$(oc get pvc -l hive.openshift.io/cluster-deployment-name=$1 -o jsonpath="{.items[0].metadata.name}")
+echo "Found PVC for cluster deployment: $PVC_NAME"
+
+echo "apiVersion: v1
+kind: Pod
+metadata:
+  name: $POD_NAME
+spec:
+  volumes:
+  - name: logsvol
+    persistentVolumeClaim:
+      claimName: $PVC_NAME
+  containers:
+  - image: centos:7
+    command: ["/bin/sh"]
+    args: ["-c", "while true\; do sleep 100000000\; done"]
+    imagePullPolicy: Always
+    name: logextractor
+    resources: {}
+    volumeMounts:
+    - mountPath: "/logs"
+      name: logsvol
+    terminationMessagePath: /dev/termination-log
+    terminationMessagePolicy: File" | oc apply -f -
+
+
+echo "waiting for $POD_NAME pod to be ready"
+
+oc wait --for=condition=Ready --timeout=120s pod/$POD_NAME
+
+echo "copying logs to $2"
+oc cp $POD_NAME:/logs $2
+
+echo "deleting logextractor pod"
+oc delete pod $POD_NAME
```

Comment:
The script should delete the pod even if there was a failure waiting for the pod to be ready or copying the logs.